web/src/features/modals/LayersModal.tsx

```diff
@@ -0,0 +1,141 @@
+  const [isLoadingLayer, setIsLoadingLayer] = useAtom(weatherButtonMap[type].loadingAtom);
+  const isEnabled = enabled === ToggleOptions.ON;
+  const Icon = weatherButtonMap[type].icon;
+  const typeAsTitlecase = type.charAt(0).toUpperCase() + type.slice(1);
+  const isDarkModeEnabled = useDarkMode();
+  const onToggle = (newEnabled: boolean) => {
+    if (newEnabled) {
+      setIsLoadingLayer(true);
+    }
+    setEnabled(newEnabled ? ToggleOptions.ON : ToggleOptions.OFF);
+  };
+
+  return (
+    <div className="flex w-full items-center justify-between p-3">
+      <div className="flex items-center">
+        <Icon size={20} className="mr-2 text-secondary dark:text-secondary-dark" />
+        <span className="text-sm font-medium text-secondary dark:text-secondary-dark">
+          {t(`${typeAsTitlecase} layer`)}
+        </span>
+      </div>
+
+      <div className="relative">
+        <SwitchToggle
+          isEnabled={isEnabled}
+          onChange={!allowed || isLoadingLayer ? () => {} : onToggle}
+          className={` ${
+            !allowed || isLoadingLayer ? 'cursor-not-allowed opacity-50' : ''
+          }`}
+          ariaLabel={t(`Toggle ${typeAsTitlecase} layer`)}
+        />
+        {isLoadingLayer && (
+          <span className="absolute inset-0 mb-[6px] ml-4 flex items-center justify-center">
+            <MoonLoader size={10} color={isDarkModeEnabled ? 'white' : 'black'} />
+          </span>
+        )}
+      </div>
+    </div>
+  );
+}
+
+export function LayersModalContent() {
+  const areWeatherLayersAllowed = useAtomValue(areWeatherLayersAllowedAtom);
+
+  return (
+    <div className="p-2">
+      <WeatherToggleSwitch allowed={areWeatherLayersAllowed} type="wind" />
+      <WeatherToggleSwitch allowed={areWeatherLayersAllowed} type="solar" />
+      {!areWeatherLayersAllowed && (
+        <p className="px-4 py-2 text-sm text-secondary dark:text-secondary-dark">
+          Weather data not available for this aggregation, switch to real-time to see live
+          weather data
+        </p>
+      )}
+    </div>
+  );
+}
+
+export default function LayersModal() {
+  const [isOpen, setIsOpen] = useAtom(isLayersModalOpenAtom);
+  const modalReference = useRef<HTMLDivElement>(null);
+  const { t } = useTranslation();
+
+  // Handle click outside to close modal
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      // Check if the event target is a button with layers-button data-testid
+      const isLayersButton = (event.target as Element)?.closest(
+        '[data-testid="layers-button"]'
+      );
+
+      // Don't close if clicking the layers button - let the toggle handler manage it
+      if (isLayersButton) {
+        return;
+      }
+
+      // Check if the click is inside the modal
+      if (
+        modalReference.current &&
+        modalReference.current.contains(event.target as Node)
+      ) {
+        return;
+      }
```

Comment:
```suggestion
      if (
        modalReference.current?.contains(event.target as Node)
      ) {
        return;
      }
```
This should be safe right?